eng/pipelines/templates/jobs/new.smoke.tests.yml

```diff
@@ -0,0 +1,179 @@
+                                  -Xpath //ItemGroup/PackageReference/@Include).Node.Value
+              if ($deps.Contains("${{ parameters.Artifact.name }}")) {
+                Write-Host "Smoke tests will run for ${{ parameters.Artifact.name }}"
+                Write-Host "##vso[task.setvariable variable=RunSmokeTests]true"
+              } else {
+                Write-Host "Smoke tests will NOT run for ${{ parameters.Artifact.name }}"
+              }
+            name: check_smoke_tests_${{ parameters.Artifact.safeName }}
+            displayName: Check smoke test eligibility for ${{ parameters.Artifact.name }}
+        - pwsh: |
+            Write-Host "Setting RunSmokeTests to ${env:RunSmokeTests}"
+            Write-Host "##vso[task.setvariable variable=RunSmokeTests;isOutput=true]${env:RunSmokeTests}"
+          name: output_eligibility
+          env:
+            RunSmokeTests: $(RunSmokeTests)
+
+  - job: SmokeTest
+    displayName: Run Smoke Test
+    ${{ if eq(parameters.Daily, false) }}:
+      dependsOn: smoke_test_eligibility
+      condition: and(succeeded(), eq(dependencies.smoke_test_eligibility.outputs['output_eligibility.RunSmokeTests'], true))
+    strategy:
+      matrix:
+        Linux (AzureCloud):
+          Pool: "azsdk-pool-mms-ubuntu-2004-general"
+          OSVmImage: "MMSUbuntu20.04"
+          TestTargetFramework: netcoreapp3.1
+          SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+          ArmTemplateParameters: $(azureCloudArmParameters)
+          Location: "westus"
+        Windows_NetCoreApp (AzureCloud):
+          Pool: "azsdk-pool-mms-win-2019-general"
+          OSVmImage: "MMS2019"
+          TestTargetFramework: netcoreapp3.1
+          SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+          ArmTemplateParameters: $(azureCloudArmParameters)
+          Location: "westus"
+        Windows_NetFramework (AzureCloud):
+          Pool: "azsdk-pool-mms-win-2019-general"
+          OSVmImage: "MMS2019"
+          TestTargetFramework: net461
+          SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+          ArmTemplateParameters: $(azureCloudArmParameters)
+          Location: "westus"
+        MacOs (AzureCloud):
+          Pool: Azure Pipelines
+          OSVmImage: "macOS-10.15"
+          TestTargetFramework: netcoreapp3.1
+          SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+          ArmTemplateParameters: $(azureCloudArmParameters)
+          Location: "westus"
+        Windows_Net50 (AzureCloud):
+          Pool: "azsdk-pool-mms-win-2019-general"
+          OSVmImage: "MMS2019"
+          TestTargetFramework: net6.0
+          SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+          ArmTemplateParameters: $(azureCloudArmParameters)
+          Location: "westus"
+        Windows_NetCoreApp (AzureUSGovernment):
```

Comment:
We will want to clean-up these matrix entries because we are not deploying live test resources any longer so we don't need to test in different clouds.